core/types/src/account.rs

```diff
@@ -0,0 +1,163 @@
+// Copyright (c) 2018-2022 The MobileCoin Foundation
+
+//! Mobilecoin Account and Subaddress objects
+
+use zeroize::Zeroize;
+
+use crate::keys::{
+    RootSpendPrivate, RootSpendPublic, RootViewPrivate, RootViewPublic, SubaddrSpendPrivate,
+    SubaddrSpendPublic, SubaddrViewPrivate, SubaddrViewPublic,
+};
+
+/// An object which represents a subaddress, and has RingCT-style
+/// view and spend public keys.
+pub trait RingCtAddress {
+    /// Get the subaddress' view public key
+    fn view_public_key(&self) -> SubaddrViewPublic;
+    /// Get the subaddress' spend public key
+    fn spend_public_key(&self) -> SubaddrSpendPublic;
+}
+
+/// Mobilecoin basic account object.
+///
+/// Typically derived via slip10, and containing root view and spend private
+/// keys.
+#[derive(Debug, Zeroize)]
+pub struct Account {
+    /// Root view private key
+    view_private: RootViewPrivate,
+    /// Root spend private key
+    spend_private: RootSpendPrivate,
+}
+
+impl Account {
+    /// Create an account from existing private keys
+    pub fn new(view_private: RootViewPrivate, spend_private: RootSpendPrivate) -> Self {
+        Self {
+            view_private,
+            spend_private,
+        }
+    }
+
+    /// Fetch account view public key
+    pub fn view_public_key(&self) -> RootViewPublic {
+        RootViewPublic::from(&self.view_private)
+    }
+
+    /// Fetch account spend public key
+    pub fn spend_public_key(&self) -> RootSpendPublic {
+        RootSpendPublic::from(&self.spend_private)
+    }
+
+    /// Fetch account view private key
+    pub fn view_private_key(&self) -> &RootViewPrivate {
+        &self.view_private
+    }
+
+    /// Fetch account spend private key
+    pub fn spend_private_key(&self) -> &RootSpendPrivate {
+        &self.spend_private
+    }
+}
+
+/// Mobilecoin private subaddress object
+#[derive(Clone, Debug, PartialEq)]
+pub struct SpendSubaddress {
```

Comment:
:wrench: I may not be up on the terms, but the description here says "private subaddress object" and there is a `Publicsubaddress`, should this be `PrivateSubaddress`?

Reply:
ahh, i could improve the comments here. afaik:

- `PublicSubaddress` has public view and spend keys and is a target for transactions
- `ViewSubaddress` has the public spend and private view keys so you can _view_ a transaction
- `SpendSubaddress` as private spend and view keys so you can _spend_ a transaction.

so i don't think `PrivateSubaddress` makes sense in this context. this is based on nomenclature used in hw support discussions, i am more than happy to be corrected if there are better descriptors.

Reply:
you probably have more insight to the naming here. 
I would suggest updating the comment to something like `/// MobileCoin spend subaddress object`